core/src/worker/workflow/machines/upsert_search_attributes_state_machine.rs

```diff
@@ -65,21 +126,39 @@ pub(super) struct Created {}
 /// higher-level machinery, it transitions into this state.
 #[derive(Debug, Default, Clone, derive_more::Display)]
 pub(super) struct CommandIssued {}
+pub(super) struct CmdRecDat {
+    sa_map: Option<SearchAttributes>,
+    replaying: bool,
+}
+
+impl CommandIssued {
+    pub(super) fn on_command_recorded(
+        self,
+        shared: SharedState,
+        dat: CmdRecDat,
+    ) -> UpsertSearchAttributesMachineTransition<Done> {
+        if shared.internal_flags.borrow_mut().try_use(
+            CoreInternalFlags::UpsertSearchAttributeOnPatch,
+            !dat.replaying,
+        ) {
+            let sa = dat.sa_map.unwrap_or_default();
+            if !shared.should_skip_determinism && shared.sa_map != sa {
```

Comment:
Was a little concerned about this equality check, but I can't currently think of a situation where byte-for-byte value check is harmful (we were bitten in Go by [this commit](https://github.com/protocolbuffers/protobuf-go/commit/582ab3de426ef0758666e018b422dd20390f7f26) which randomly adds whitespace to prevent proto byte checks, but this is checking json-encoded byte arrays not proto-encoded byte arrays so it should be ok).